ProcessLib/HT/HTFEM.h

```diff
@@ -88,6 +88,76 @@ class HTFEM : public HTLocalAssemblerInterface
         return Eigen::Map<const Eigen::RowVectorXd>(N.data(), N.size());
     }
 
+    /// Computes the flux in the point \c pnt_local_coords that is given in
+    /// local coordinates using the values from \c local_x.
+    // TODO add time dependency
+    std::vector<double> getFlux(
+        MathLib::Point3d const& pnt_local_coords,
+        std::vector<double> const& local_x) const override
+    {
+        // eval dNdx and invJ at given point
+        using FemType =
+            NumLib::TemplateIsoparametric<ShapeFunction, ShapeMatricesType>;
+
+        FemType fe(*static_cast<const typename ShapeFunction::MeshElement*>(
+            &this->_element));
+
+        typename ShapeMatricesType::ShapeMatrices shape_matrices(
+            ShapeFunction::DIM, GlobalDim, ShapeFunction::NPOINTS);
+
+        // Note: Axial symmetry is set to false here, because we only need dNdx
+        // here, which is not affected by axial symmetry.
+        fe.computeShapeFunctions(pnt_local_coords.getCoords(), shape_matrices,
+                                 GlobalDim, false);
+        std::vector<double> flux;
+        flux.resize(3);
+
+        // fetch permeability, viscosity, density
+        SpatialPosition pos;
+        pos.setElementID(this->_element.getID());
+
+        MaterialLib::Fluid::FluidProperty::ArrayType vars;
+        double T_int_pt = 0.0;
+        double p_int_pt = 0.0;
+
+        // local_x contains the local temperature and pressure values
+        NumLib::shapeFunctionInterpolate(local_x, shape_matrices.N, T_int_pt, p_int_pt);
+
+        // set the values into array vars
+        vars[static_cast<int>(MaterialLib::Fluid::PropertyVariableType::T)] =
+            T_int_pt;
+        vars[static_cast<int>(MaterialLib::Fluid::PropertyVariableType::p)] =
+            p_int_pt;
+
+        // TODO remove following line if time dependency is implemented
+        double const t = 0.0;
+        auto const K = this->_material_properties.porous_media_properties
+                           .getIntrinsicPermeability(t, pos)
+                           .getValue(t, pos, 0.0, T_int_pt);
+
+        auto const mu = this->_material_properties.fluid_properties->getValue(
+            MaterialLib::Fluid::FluidPropertyType::Viscosity, vars);
+        GlobalDimMatrixType const K_over_mu = K / mu;
+
+        auto const p_nodal_values = Eigen::Map<const NodalVectorType>(
+            &local_x[local_x.size()/2], ShapeFunction::NPOINTS);
+        Eigen::RowVectorXd q =
+            -K_over_mu * shape_matrices.dNdx * p_nodal_values;
+
+        if (this->_material_properties.has_gravity)
+        {
+            auto const rho_w =
+                this->_material_properties.fluid_properties->getValue(
+                    MaterialLib::Fluid::FluidPropertyType::Density, vars);
+            auto const b = this->_material_properties.specific_body_force;
+            // here it is assumed that the vector b is directed 'downwards'
+            q += K_over_mu * rho_w * b;
```

Comment:
I'm not getting the intentions of the comment. What is 'downwards'? Maybe it's copied from some other context...